src/cache/cache_las.c

```diff
@@ -1237,9 +1238,21 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
         /*
          * Remove the previous obsolete entries whenever the scan moves into next key or the current
          * LAS record is not a modify record.
+         *
+         * The special ondisk value i.e both timestamp and transaction id values as 0 and is the
+         * last record of the key that is globally visible is present in the LAS can only be removed
+         * whenever a next record of the same key gets removed. This is to protect the cases where
+         * we may still need the older ondisk value for some scenarios. The draw back is that this
+         * record never be removed until more updates happens on that key, but these scenarios are
+         * minimal.
+         *
+         * TODO: Better key/value pair design that let you know that on disk image of the key is
+         * globally visible will simplify the logic of removing the entire LAS records for that key.
          */
-        if (pending_remove_cnt && (key_change || upd_type != WT_UPDATE_MODIFY)) {
+        if ((special_ondisk_value ? pending_remove_cnt > 1 : pending_remove_cnt > 0) &&
```

Comment:
Just to clarify, is this because if we have a globally visible update we don't want to count that one? We need to have at least one more update getting removed before sweep is allowed to remove it?

Reply:
Las sweep server has no information to identify the globally visible on disk value even if there is a disk image that is globally visible. So it will be removed only whenever any LAS record of the same key gets removed.